src/ids/node.rs

```diff
@@ -168,11 +169,78 @@ impl<'de> Deserialize<'de> for Id {
     where
         D: Deserializer<'de>,
     {
-        let s: &str = Deserialize::deserialize(deserializer)?;
+        struct StringOrStr;
+
+        impl<'de> Visitor<'de> for StringOrStr {
+            type Value = Cow<'de, str>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a borrowed or owned string")
+            }
+
+            fn visit_string<E>(self, v: String) -> std::result::Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(Cow::Owned(v))
+            }
+
+            fn visit_borrowed_str<E>(self, v: &'de str) -> std::result::Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Ok(Cow::Borrowed(v))
+            }
+        }
+
+        let s = deserializer.deserialize_any(StringOrStr)?;
         Id::from_str(&s).map_err(serde::de::Error::custom)
     }
 }
 
+/// RUST_LOG=debug cargo test --package avalanche-types --lib -- ids::node::test_custom_de_serializer --exact --show-output
+#[test]
+fn test_custom_de_serializer() {
+    #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+    struct Data {
+        node_id: Id,
+    }
+
+    let d = Data {
+        node_id: Id::from_str("NodeID-6ZmBHXTqjknJoZtXbnJ6x7af863rXDTwx").unwrap(),
+    };
+
+    let yaml_encoded = serde_yaml::to_string(&d).unwrap();
+    println!("yaml_encoded:\n{}", yaml_encoded);
+    let yaml_decoded = serde_yaml::from_str(&yaml_encoded).unwrap();
+    assert_eq!(d, yaml_decoded);
+
+    let json_encoded = serde_json::to_string(&d).unwrap();
+    println!("json_encoded:\n{}", json_encoded);
+    let json_decoded = serde_json::from_str(&json_encoded).unwrap();
+    assert_eq!(d, json_decoded);
+
+    let json_decoded_2: Data = serde_json::from_str(
+        "
+
+{
+    \"node_id\":\"NodeID-6ZmBHXTqjknJoZtXbnJ6x7af863rXDTwx\"
+}
+
+",
+    )
```

Comment:
I've got another PR coming that converts the other raw-strings to string-literals. 
```suggestion
    let json_decoded_2: Data = serde_json::from_str(r#"{ "node_id":"NodeID-6ZmBHXTqjknJoZtXbnJ6x7af863rXDTwx" }"#)
```
Please don't actually commit directly here as `cargo fmt` might have ideas about where that `.unwrap()`  goes.